### DISCUSSION TITLE
ModuleNotFoundError: No module named 'telegram.utils'

### DISCUSSION THREAD

**Discussion by joker096** [python-telegram-bot/python-telegram-bot#4124] [Q&A]
File "...posts.py", line 6, in <module>
    from telegram.utils.request import Request

```python
from telegram.utils.request import Request
class PostBot:
    def __init__(self, bot_token, chat_id):
        init_firebase()
        # request = Request(con_pool_size=8)
        self.bot = Bot(token=bot_token, request=request)
        self.chat_id = chat_id
        self.like_count = 0
        self.dislike_count = 0
        self.user_data = {}   
```

python-telegram-bot         20.8  - when pip list

How to fix problem? Tried to uninstall and install again etc.

**Comment by joker096** [comment]
Name: python-telegram-bot
Version: 20.8
Summary: We have made you a wrapper you can't refuse
Home-page: https://python-telegram-bot.org/
Author: Leandro Toledo
Author-email: devs@python-telegram-bot.org
License: LGPLv3
Location: \.venv\Lib\site-packages
Requires: httpx
Required-by:

**Comment by Bibo-Joshi** [comment]
You have installed v20.8 but your code is written for <=v13.15. you'll have to update your code. See https://github.com/python-telegram-bot/python-telegram-bot/wiki/Transition-guide-to-Version-20.0.

**Reply by joker096** [reply]
Problem is in streamlit - it doesn`t support async functions. how to use Request anyway?

**Comment by joker096** [comment]
I`ve downloaded version - the same problem exists.

**Comment by joker096** [comment]
If I remove Request and its import from your library  - all code works fine, but I can`t send meesage to Telegram...

**Comment by joker096** [comment]
```python
import streamlit as st
from telegram import Bot, InlineKeyboardButton, InlineKeyboardMarkup
from telegram.ext import Updater, CallbackQueryHandler, MessageHandler, filters
from telegram.request import RequestData
import firebase_admin
from firebase_admin import credentials, db
import logging
from firebase_utils import init_firebase, init_session_state
import sys


class PostBot:
    def __init__(self, bot_token, chat_id):
        init_firebase()
        request = RequestData()
        self.bot = Bot(token=bot_token, request=request)
        self.chat_id = chat_id
        self.like_count = 0
        self.dislike_count = 0
        self.user_data = {}   
```

now problem in 
An error occurred while sending the post: 'coroutine' object has no attribute 'chat'

**Comment by Bibo-Joshi** [comment]
Please

* Send one comment instead of 5
* Properly format your code as code block
* Instead of sending only the error message, send the complete traceback, formatted as code block
* Search "python asyncio tutorial" in your favorite search engine to get familiar with asyncio
* Consider using asyncio friendly libraries
